internal/pkg/utils/testhelper/storageenvmock/storageenvmock.go

```diff
@@ -0,0 +1,36 @@
+package storageenvmock
```

Comment:
Ještě se zbavuju závislosti na Storage klientovi v operaci.

Reply:
Ok. Osobne by som zacal s tym, ze by som spravil z `ticketProvider` zavislost.

Namiesto `storageapi.NewTicketProvider(ctx, storageApiClient)` by sa volalo `tickerProviderFactory.New(ctx)`.

A do danych `dependencies` interface, by sa pridalo `TicketProviderFactory() TicketProviderFactory`.

```go
type TicketProviderFactory interface {
  New(ctx context.Context) TicketProvider
}

type TicketProvider interface {
  Request(onSuccess func(ticket *Ticket))
  Resolve() error
}
```

`TicketProvider` je teraz `struct`, bol by to interface.
Ktory by mal 2 implementacie `ApiTicketProvider` (aktualna implementaci), `Sequence/TestTicketProvider`

----- 

Vysledkom by bolo, ze nakoniec by ti z template uplne vypadla zavslost na `StorageApi`, a zostala iba zavislost na `TicketProviderFactory`.

----

^^^ toto sa da pekne cele spravit v samostatnom PR, v `go-client` a tu v KAC.

Reply:
🤔 To myslíš namísto tohoto commitu? V tuto chvíli už tam závislost Storage klienta přímo nepoužívám. 

Zbývá mi dořešit, že když do `opts` dependencies pro projekt nastavím blbost, tak někde zevnitř `use template` operace dostanu chybu. Ještě jsem nepřišel na to odkud. 🙂 Tímhle by se to vyřešilo?

https://github.com/keboola/keboola-as-code/blob/943836d24ece4cacd68bbe3000441b92bad63f3d/pkg/lib/operation/template/local/test/operation.go#L112-L115

![CleanShot 2022-07-26 at 14 24 41@2x](https://user-images.githubusercontent.com/215660/181005547-91f194ac-e8f1-4dda-97c2-209c0224fecb.jpg)

Reply:
no ked sa so sablony vytvaraju objekty v projekte, tak sa im generuju nove IDcka, tak to je to co ja popisoval, a imho aj tvoj problem .... generovanie ticketov v teste je asi to co si riesil ty, ... tam by to vyriesil tiez ten interface ... a ta testovacia implementacia, ktoru vlastne uz asi mas ... pozrieme na calle .... idem si dochystat tasky